File: hw/Rodriguez/u2/hw18/Zoo/ec_edu_espe_zoo_model/condor.py
from .bird import Bird
from datetime import date
from .cage import Cage

class Condor(Bird):
    def __init__(self, mountains, id, description, date_onborn, gender, number_of_bones, cage):
        super().__init__(id, description, date_onborn, gender, number_of_bones, cage)
        self.mountains = mountains

    def __str__(self):
        return f"Condor(mountains={self.mountains}, {super().__str__()})"

    def feed(self, food):
        raise NotImplementedError("Subclasses should implement this method")

File: hw/ccedeno/u2/HW18ZooInPythonJavaScript/ZooInJavaScript/ec.edu.espe.Zoo.model/Condor.js
const Bird = require('../ec.edu.espe.Zoo.model/Bird');

class Condor extends Bird {
  constructor(mountains, id, description, dateOnborn, gender, numberOfBones, cage) {
    super(id, description, dateOnborn, gender, numberOfBones, cage);
    this.mountains = mountains;
  }

  toString() {
    return `Condor{mountains=${this.mountains}, ${super.toString()}}`;
  }

  feed(food) {
    throw new Error('Method "feed" is not supported yet.');
  }

  // Getters and Setters
  getMountains() {
    return this.mountains;
  }

  setMountains(mountains) {
    this.mountains = mountains;
  }
}

module.exports = Condor;
